builtin/v13/check.go

```diff
@@ -288,6 +288,42 @@ func CheckDealStatesAgainstSectors(acc *builtin.MessageAccumulator, minerSummari
 		acc.Require(deal.SlashEpoch <= sectorDeal.SectorExpiration,
 			"deal state slashed at %d after sector expiration %d for miner %v",
 			deal.SlashEpoch, sectorDeal.SectorExpiration, deal.Provider)
+
+		acc.Require((deal.SectorNumber == sectorDeal.SectorNumber) || (deal.SectorNumber == 0 && deal.SlashEpoch != -1),
+			"deal sector number %d does not match sector %d for miner %v (ds: %#v; ss %#v)",
+			deal.SectorNumber, sectorDeal.SectorNumber, deal.Provider, deal, sectorDeal)
+	}
+
+	// HAMT[ActorID]HAMT[SectorNumber]SectorDealIDs
+	marketDealToSector := make(map[abi.DealID]abi.SectorID)
+
+	for sectorID, dealIDs := range marketSummary.ProviderSectors {
+		maddr, err := address.NewIDAddress(uint64(sectorID.Miner))
+		if err != nil {
+			acc.Addf("error creating ID address: %v", err)
+			continue
+		}
+
+		minerSummary, found := minerSummaries[maddr]
+		if !found {
+			acc.Addf("provider %v for sector %v not found among miners", sectorID, sectorID)
+			continue
+		}
+
+		sector := minerSummary.SectorsWithDeals[sectorID.Number]
```

Comment:
@anorth If you could chime in here, that would be great.

Reply:
Regarding making it "correct" to start with, I do somewhat strongly want to keep as much migration-specific invariants here, it'll make testing the migration significantly easier, and that is very important IMO.

Reply:
Hm. Yeah, you're right. However, I still think going from miner -> market is better than going from market -> miner because it'll help us identify deals that haven't been migrated.

Reply:
But IIUC we don't want every deal found in miner data to be in `ProviderSectors`, only the specific subset as detailed [here](https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0076.md#migration). I could implement that very specific check, but at that point the "invariant" is just rewriting the migration, I think.

Reply:
Ah... you're right. We don't have a 1:1 mapping anyways.